okhttp/src/main/java/okhttp3/Headers.java

```diff
@@ -188,9 +186,7 @@ public static Headers of(Map<String, String> headers) {
       }
       String name = header.getKey().trim();
       String value = header.getValue().trim();
-      if (name.length() == 0 || name.indexOf('\0') != -1 || value.indexOf('\0') != -1) {
-        throw new IllegalArgumentException("Unexpected header: " + name + ": " + value);
-      }
+      checkNameAndValue(name, value);
```

Comment:
It seems like this is right, stuff should use Builder.addLenient() if they don't want validation.  But was there another reason not to fully validate?